docs/deep-dives/esm.md

```diff
@@ -160,12 +160,64 @@ We will apply a community codemod to help with the transition, but it will likel
 
 ---
 
+## Future considerations
+
+TypeScript 5.7 introduced [`rewriteRelativeImportExtensions`], which allows us to rewrite relative import paths to include the `.js` file extension. When combined with [`allowImportingTsExtensions`], this would allow us to import `.ts` files directly.
+
+Node.js also recently added [native type stripping support], which means we could start using Node.js to run TypeScript code directly without compilation.
```

Comment:
FWIW I'm not sure this should be a goal for production applications, especially if/when we go harder on bundling and tree shaking. But it would simplify local run, REPL, scripting, etc.